etcd/etcd2-live-http-to-https-migration.md

```diff
@@ -0,0 +1,192 @@
+# etcd2 Live HTTP to HTTPS migration
+
+If you already have existing etcd2 cluster which uses insecure HTTP connections and you would like to switch from HTTP to HTTPS and in addition enable TLS certificate authentication, you have to follow instruction below.
+
+## Prepare Cluster Components
+
+### Check insecure port availability
+
+By default fleet and flannel use two etcd client ports: 2379 and 4001. Last one is used for etcd v0.4 backward compatibility. In this example we will use 4001 TCP port for insecure HTTP connections and our etcd members will listen that port only on local interface for security reason. This will allow us to switch transparently to HTTPS configuration without services downtime and simplify configuration
+
+If you've configured flannel, fleetd or other local components to use custom or 2379 port only, you have to configure them to use 4001 TCP port.
+
+If your etcd2 configuration doesn't have configured 4001 client port, you have to enable it before we proceed with this manual. For example you use Cloud-Config to configure your CoreOS cluster. In this case you have to retrieve `ETCD_LISTEN_CLIENT_URLS` value from `/run/systemd/system/etcd2.service.d/20-cloudinit.conf` config file:
+
+```sh
+grep ETCD_LISTEN_CLIENT_URLS /run/systemd/system/etcd2.service.d/20-cloudinit.conf
+Environment="ETCD_LISTEN_CLIENT_URLS=http://0.0.0.0:2379"
+```
+
+Then create `/etc/systemd/system/etcd2.service.d/` directory if it doesn't exist and create [drop-in][drop-ins] file: `/etc/systemd/system/etcd2.service.d/25-insecure_localhost.conf`. Use value we've retrieved above and append `http://127.0.0.1:4001` URL at the end:
+
+```
+[Service]
+Environment="ETCD_LISTEN_CLIENT_URLS=http://0.0.0.0:2379,http://127.0.0.1:4001"
+```
+
+Then run `systemctl daemon-reload` and `systemctl restart etcd2` to restart etcd2. Check etcd2 availability using these commands:
+
+```sh
+etcdctl member list
```

Comment:
It's good to get people to verify that their cluster is working properly before we start messing with it. I wonder if this should be more towards the top.